### PR TITLE
Move platform-specific pieces of `CertificatePal` into separate classes.

### PR DESCRIPTION
**Pull request by baulig**
* Pal.OSX/CertificatePal.cs: Move `AppleCertificatePal` into its own file and add
  all static methods from `CertificatePal` to it, turning `CertificatePal` into a
  tiny proxy.

* Pal.Unix/CertificatePal.cs: Rename into `OpenSslCertificatePal` to make it consistent
  with OS X, then add `CertificatePal` proxy.

* Pal.Unix/OpenSslX509Encoder.cs and StorePal.cs: use `OpenSslCertificatePal` instead
  of `CertificatePal`.

### CODE REVIEW

**Comment by filipnavara** [comment]
This generally makes sense to me, but I would expect the Windows PAL to follow the convention too.

File: src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslCertificatePal.cs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+using System;
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal sealed partial class OpenSslCertificatePal
```
Comment (bartonjs) [inline src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslCertificatePal.cs]: All of the methods added here should just go on OpenSslCertificateReader.  The macOS and Windows types that end in "CertificatePal" are the ICertificatePal implementations, having a "-CertificatePal" type here that isn't is more confusing than it just having a different name.

**Comment by bartonjs** [comment]
Looks like this conflicted with putting Export on ICertificatePal.

**Comment by baulig** [comment]
Should be fixed now, I rebased on latest master.

**Comment by bartonjs** [comment]
@dotnet-bot Test Linux-musl x64 Debug Build please

**Pull request merged**